src/main/java/org/checkerframework/checker/noliteral/qual/MaybeConstant.java

```diff
@@ -0,0 +1,36 @@
+package org.checkerframework.checker.noliteral.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.DefaultInUncheckedCodeFor;
+import org.checkerframework.framework.qual.LiteralKind;
+import org.checkerframework.framework.qual.QualifierForLiterals;
+import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.TypeUseLocation;
+
+/**
+ * The top type for the NoLiteral Checker's type system.
+ *
+ * <p>This type represents an expression that might contain a literal, like 5 or {0xa, 0xb}. It is
```

Comment:
"expression contains" is odd phrasing, since it sounds local and syntactic.  I think you mean something more like it could be derived/computed (in part) from a literal.  In this case, `@MaybeConstant` is a somewhat confusing name for the concept, since `"hello" + readInput()` is definitely not constant (from a user perspective) but has type `@MaybeConstant`.

Reply:
I've renamed it to `MaybeDerivedFromConstant`, which I think is much clearer. Thanks for pointing this out.